homeassistant/components/lg_netcast/media_player.py

```diff
@@ -79,19 +107,39 @@ class LgTVDevice(MediaPlayerEntity):
     _attr_assumed_state = True
     _attr_device_class = MediaPlayerDeviceClass.TV
     _attr_media_content_type = MediaType.CHANNEL
+    _attr_has_entity_name = True
+    _attr_name = None
 
-    def __init__(self, client, name, on_action_script):
+    def __init__(self, client, name, model, *, unique_id=None):
         """Initialize the LG TV device."""
         self._client = client
-        self._name = name
         self._muted = False
-        self._on_action_script = on_action_script
+        self._turn_on = PluggableAction(self.async_write_ha_state)
         self._volume = 0
         self._channel_id = None
         self._channel_name = ""
         self._program_name = ""
         self._sources = {}
         self._source_names = []
+        if unique_id is not None:
+            self._attr_unique_id = unique_id
+            self._attr_device_info = DeviceInfo(
+                identifiers={(DOMAIN, unique_id)},
+                manufacturer=ATTR_MANUFACTURER,
+                name=name,
+                model=model,
+            )
+
+    async def async_added_to_hass(self) -> None:
+        """Connect and subscribe to dispatacher signals and state updates."""
+        await super().async_added_to_hass()
+
+        if (entry := self.registry_entry) and entry.device_id:
```

Comment:
we know this entity has a unique id, hence there's no need to check this

Reply:
Sure, I think this is mainly here to keep the type checker happy (and is the same structure as the samgsung and webostv and webostv components) but I can instead turn this into a `TYPING_CHECKING` assert check instead.

Reply:
Done